src/core/blocks/casting.cpp

```diff
@@ -242,278 +242,113 @@ template <CBType FROMTYPE> struct ToBytes {
-  };                                                                           \
-  RUNTIME_CORE_BLOCK(To##_varName_);                                           \
-  RUNTIME_BLOCK_inputTypes(To##_varName_);                                     \
-  RUNTIME_BLOCK_outputTypes(To##_varName_);                                    \
-  RUNTIME_BLOCK_compose(To##_varName_);                                        \
-  RUNTIME_BLOCK_activate(To##_varName_);                                       \
-  RUNTIME_BLOCK_END(To##_varName_);
-
-TO_SOMETHING_SIMPLE(Int, int64_t, intValue, stoll, IntType);
-TO_SOMETHING(Int, 2, int64_t, int2Value, stoll, Int2Type);
-TO_SOMETHING(Int, 3, int32_t, int3Value, stol, Int3Type);
-TO_SOMETHING(Int, 4, int32_t, int4Value, stol, Int4Type);
-TO_SOMETHING_SIMPLE(Float, double, floatValue, stod, FloatType);
-TO_SOMETHING(Float, 2, double, float2Value, stod, Float2Type);
-TO_SOMETHING(Float, 3, float, float3Value, stof, Float3Type);
-TO_SOMETHING(Float, 4, float, float4Value, stof, Float4Type);
+template <CBType ToType> struct ToNumber {
+  const VectorTypeTraits *_outputVectorType{nullptr};
+  const NumberTypeTraits *_outputNumberType{nullptr};
+
+  CBTypesInfo inputTypes() { return CoreInfo::AnyType; }
+  CBTypesInfo outputTypes() {
+    if(!_outputVectorType)
+      return CoreInfo::AnyType;
+    return _outputVectorType->type; 
+  }
+  
+  CBTypeInfo compose(const CBInstanceData &data) {
+    _outputVectorType =
+        VectorTypeLookup::getInstance().get(ToType);
+    if (!_outputVectorType) {
+      throw ComposeError("Conversion not implemented for this type");
+    }
+
+    _outputNumberType = NumberTypeLookup::getInstance().get(_outputVectorType->numberType);
+
+    return _outputVectorType->type;
+  }
+
+  void parseVector(CBVar &output, const CBVar &input) {
+    const VectorTypeTraits *inputVectorType =
+        VectorTypeLookup::getInstance().get(input.valueType);
+    if (!inputVectorType) {
+      throw ActivationError("Cannot cast given input type.");
+    }
+
+    const NumberConversion *conversion =
+        NumberTypeLookup::getInstance().getConversion(
+            inputVectorType->numberType, _outputVectorType->numberType);
+    assert(conversion);
+
+    // Can reduce call overhead by adding a single ConvertMultiple function with
+    // in/out lengths
+    uint8_t *srcPtr = (uint8_t*)&input.payload;
+    uint8_t *dstPtr = (uint8_t*)&output.payload;
+    size_t numToConvert =
+        std::min(_outputVectorType->dimension, inputVectorType->dimension);
+    for (size_t i = 0; i < numToConvert; i++) {
+      conversion->convertOne(srcPtr, dstPtr);
+      srcPtr += conversion->inStride;
+      dstPtr += conversion->outStride;
+    }
+  }
+
+  void parseSeqElements(CBVar &output, const CBSeq &sequence) {
+    uint8_t *dstPtr = (uint8_t*)&output.payload;
+    for (size_t i = 0; i < sequence.len; i++) {
+      const CBVar &elem = sequence.elements[i];
+
+      const NumberTypeTraits *elemNumberType =
```

Comment:
to warmup?

Reply:
The switch is there for each element in the original TO_SOMETHING implementation.
```
 for (uint32_t i = 0; i < 1 && i < input.payload.seqValue.len; i++) {   \
          if (convert(output, input.payload.seqValue.elements[i]))             \
            return output;                                                     \
        }  
...
 bool convert(CBVar &dst, const CBVar &src) {                               \
      switch (src.valueType) {                                                 \
```
Can I check all the input element types of a seq during warmup and is it possible for them to always be known during warmup?

Reply:
The original is probably not so an accurate implementation sadly.
Yes blocks can enforce sequence types, in this case they should enforce that the input sequence is always of the same type...
But I see in this case it's not it takes `any`. 
I'm afraid that this approach is much slower than the previous one though in that case if we have to do hash and lookup. (although likely fast yet not the same of a switch)

Possible solution:
* Evaluate in compose what input types you are getting `data.inputType`
* If it's a sequence with a single type you can optimize the lookup, type won't ever change (except if Any)
* If it's a sequence with multiple types we need to do the lookup variation of the activation as we don't know at runtime